### PR TITLE
[gatsby-plugin-catch-links] fix/workaround for IE

### PR DESCRIPTION
**Pull request by ahinrichs**
Two regressions broke the catch-links plugin in Internet Explorer: 

* [PR 3996](https://github.com/gatsbyjs/gatsby/pull/3996) uses `a.pathname`. IE does not include the starting slash there (see [this stackoverflow topic](https://stackoverflow.com/questions/956233/javascript-pathname-ie-quirk)) so this check always fails.
* Second [PR 6866](https://github.com/gatsbyjs/gatsby/pull/6866) uses URL API which does not work in IE.

The PR here just adds a quirk for the `a.pathname` bug. But the second issue **must** be fixed too before merging. In this state IE will again get past the line 81 and will raise an error at `var anchoreUrl = new URL(anchor.href)`

### CODE REVIEW

File: packages/gatsby-plugin-catch-links/src/catch-links.js [hunk 1]
```diff
@@ -78,7 +78,8 @@ module.exports = function(root, cb) {
     // will navigate to an external link instead of doing a pushState resulting
     // in `https://example.com/myapp/https://example.com/not-my-app`
     var re = new RegExp(`^${a2.host}${withPrefix(`/`)}`)
-    if (!re.test(`${a1.host}${a1.pathname}`)) return true
+    var a1pathname = (a1.pathname[0] !== '/' ? '/' : '') + a1.pathname
```
Comment (pieh) [inline packages/gatsby-plugin-catch-links/src/catch-links.js]: can you add comment why this is needed (ie weird quirk)?

Comment (ahinrichs) [reply packages/gatsby-plugin-catch-links/src/catch-links.js]: See https://stackoverflow.com/questions/956233/javascript-pathname-ie-quirk. 

IE does not include the starting slash of an URL in a.pathname. So `a.href="http://example.com/page"` will get `a.pathname="/page"` in chrome et.al (and this is what the standard says) but `a.pathname="page"` in IE.

Comment (pieh) [reply packages/gatsby-plugin-catch-links/src/catch-links.js]: I mean - add comment in code - I got the gist of problem from PR description - but this is for other folks that will read this code later

Comment (ahinrichs) [reply packages/gatsby-plugin-catch-links/src/catch-links.js]: Sorry, of course I will add a comment.

I started looking for the URL API replacement fix but I wasn't sure what this code is really doing. An IE fix for this need to be added to this PR before merging.

**Comment by ahinrichs** [comment]
I found URL is only used for anchor origin--which is missing in IE anyway. So added comment and a second workaround for getting anchor and window.location origin in IE.

**Comment by pieh** [comment]
This was recently merged into v1 branch https://github.com/gatsbyjs/gatsby/pull/7779 - could this be reused? it seems pretty comprehensive there

**Comment by ahinrichs** [comment]
I don't know why I just found [#7779](https://github.com/gatsbyjs/gatsby/pull/7779) just today. But yes, it should be straight forward to combine this and include the underlying fixes of this one. I would expect that it even should be worth backporting this one to v1. Will check over weekend if not someone else will be faster.

**Comment by pieh** [comment]
There is open PR for porting that PR to v2 - https://github.com/gatsbyjs/gatsby/pull/8646 I'm not sure if it fixes ie11 problem or not

**Comment by ahinrichs** [comment]
The v1 rewrite is broken in IE. I didn't check, whether it is functionally equivalent with the last v2 commit but just included the fix for the a.pathname bug in IE. So this PR could be closed in favour of [#8646](8646)

**Pull request closed without merging**